hack/rollback_ci-operator.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+# Script to rollback the ci-operator version and mirror the image to QCI
+
+# Function to check required tools
+check_requirements() {
+    for cmd in oc yq jq; do
+        if ! command -v $cmd &> /dev/null; then
+            echo "Error: $cmd is not installed. Please install it and try again."
+            exit 1
+        fi
+    done
+}
+
+# Function to get the tags of ci-operator from app.ci
+get_tags() {
+    echo "Fetching tags from app.ci imagestream..."
+    oc --context app.ci -n ci get imagestreams ci-operator -o yaml | yq .status.tags
+    if [ $? -ne 0 ]; then
+        echo "Error: Failed to fetch tags from app.ci."
+        exit 1
+    fi
+}
+
+# Function to validate digest
+validate_digest() {
+    if [[ ! $1 =~ ^sha256:[a-f0-9]{64}$ ]]; then
+        echo "Error: Invalid digest format. Please provide a valid sha256 digest."
+        exit 1
+    fi
+}
+
+# Function to tag the given digest to ci-operator:latest
+tag_digest() {
+    local digest=$1
+    echo "Tagging ci-operator@$digest to ci-operator:latest..."
+    oc --context app.ci -n ci tag ci-operator@$digest ci-operator:latest
+    if [ $? -ne 0 ]; then
+        echo "Error: Failed to tag the digest to ci-operator:latest."
+        exit 1
+    fi
+}
+
+# Function to mirror the ci-operator:latest image to QCI
+mirror_image() {
+    local creds_file=$(mktemp /tmp/registry-push-credentials.XXXXXX.c)
+    trap "rm -f $creds_file" EXIT  # Ensure creds file is removed
```

Comment:
Credentials forgotten somewhere on the filesystem concerns me a bit. Does `trap ...` work even when the scripts raises an unexpected error?